static/js/services/delete-doc.js

```diff
@@ -8,6 +8,9 @@
     if (doc.type === 'person' && doc.parent && doc.parent._id) {
       db.getDoc(doc.parent._id, function(err, parent)  {
         if (err) {
+          if (err.reason === "deleted") {
+            return callback();
+          }
```

Comment:
Sure the atomic bit may always be an issue. But test out (a) and if it's not working raise an issue so it's covered.